events/equeue/equeue.c

```diff
@@ -601,23 +601,27 @@ static void equeue_chain_update(void *p, int ms)
     if (ms >= 0) {
         c->id = equeue_call_in(c->target, ms, equeue_chain_dispatch, c->q);
     } else {
-        equeue_dealloc(c->target, c);
+        equeue_dealloc(c->q, c);
     }
 }
 
-void equeue_chain(equeue_t *q, equeue_t *target)
+int equeue_chain(equeue_t *q, equeue_t *target)
 {
     if (!target) {
         equeue_background(q, 0, 0);
-        return;
+        return 0;
     }
 
     struct equeue_chain_context *c = equeue_alloc(q,
                                                   sizeof(struct equeue_chain_context));
+    if (!c) {
+        return -1;
```

Comment:
How about returning a real error code such as MBED_ERROR_FAILED_OPERATION, you can also define new a specific error code if needed. We are changing the API anyway, so why not make it better by adopting the error codes instead of -1. Similarly for success we can use MBED_SUCCESS(which is 0). This helps with error analytics as well.

Reply:
@SenRamakri, sounds like a good idea for a separate PR. This would be a change in the types of error codes that equeue returns and we should update all of the other error codes as well (they are all -1 currently).

Also note the only errors equeue returns are when the equeue has no memory left.

Reply:
@geky - I don't think having to change all error codes is a reason for not making the change. Also, you can use OUT_OF_MEMORY error code for no memory left situation. Irrespective of that, according to my offline discussion with @deepikabhavnani, since we are pulling this from mbed_events which can theoretically exist independently(without mbed-os) we decided to not use mbed error codes for now.

Reply:
> good idea for a separate PR

It's just out of scope for this PR

We could put the error codes in the porting definitions and ifdef it for Mbed:
https://github.com/ARMmbed/mbed-os/blob/master/events/equeue/equeue_platform.h